framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -149,304 +162,209 @@ public boolean hasDependentAnnotations() {
-     * @return true if method is a varargs method and its varargs arguments are not passed in an
-     *     array
-     */
-    private boolean isVarArgsInvocation(
-            ExecutableElement method,
-            AnnotatedExecutableType methodType,
-            List<? extends ExpressionTree> args) {
-        if (method != null && method.isVarArgs()) {
-            if (method.getParameters().size() != args.size()) {
-                return true;
-            }
-            AnnotatedTypeMirror lastArg = factory.getAnnotatedType(args.get(args.size() - 1));
-            List<AnnotatedTypeMirror> paramTypes = methodType.getParameterTypes();
-            AnnotatedArrayType lastParam =
-                    (AnnotatedArrayType) paramTypes.get(paramTypes.size() - 1);
-            return lastArg.getKind() != TypeKind.ARRAY
-                    || AnnotatedTypes.getArrayDepth(lastParam)
-                            != AnnotatedTypes.getArrayDepth((AnnotatedArrayType) lastArg);
+        StringToJavaExpression stringToJavaExpr;
+        if (tree instanceof MethodInvocationTree) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atMethodInvocation(
+                                    stringExpr, (MethodInvocationTree) tree, factory.getChecker());
+        } else if (tree instanceof NewClassTree) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atConstructorInvocation(
+                                    stringExpr, (NewClassTree) tree, factory.getChecker());
+        } else {
+            throw new BugInCF("Unexpected tree: %s kind: %s", tree, tree.getKind());
         }
-        return false;
+        convertAnnotatedTypeMirror(stringToJavaExpr, declaredMethodType);
+        this.viewpointAdaptedCopier.visit(declaredMethodType, methodType);
     }
 
-    ///
-    /// Standardization
-    ///
-
     /**
-     * Creates a TreeAnnotator that standardizes dependent type annotations.
+     * Viewpoint-adapts the Java expressions in annotations written on a field declaration for the
+     * use at {@code fieldAccess}.
      *
-     * @return a new TreeAnnotator that standardizes dependent type annotations
-     */
-    public TreeAnnotator createDependentTypesTreeAnnotator() {
-        assert hasDependentAnnotations();
-        return new DependentTypesTreeAnnotator(factory, this);
-    }
-
-    /**
-     * Standardizes the Java expressions in annotations on a constructor invocation.
-     *
-     * @param tree the constructor invocation
-     * @param type the type of the expression; is side-effected by this method
+     * @param fieldAccess a field access
+     * @param type its type; is side-effected by this method
      */
-    public void standardizeNewClassTree(NewClassTree tree, AnnotatedDeclaredType type) {
+    public void atFieldAccess(MemberSelectTree fieldAccess, AnnotatedTypeMirror type) {
```

Comment:
This is a standardization routine.  I think than starting a viewpoint-adaptation routine's name with "at" is appropriate, but for a standardization routine it is misleading and does not make a distinction between the two types of functionality.  I suggest changing the standardization routine names; even adding "standardize" at the beginning of them would be clearer.

Reply:
This is not a standardization routine.  It viewpoint-adapts annotations from the declaration of the field to the field access.  For example, `type` is `@KeyFor("this.map") List<String>` and `fieldAccess` is `myClass.field`, then this method changes the KeyFor annotation to `@KeyFor("myClass.map")`.

I'm still thinking about renaming the methods that do not viewpoint-adapt.

Reply:
>I think than starting a viewpoint-adaptation routine's name with "at" is appropriate, but for a standardization routine it is misleading and does not make a distinction between the two types of functionality. I suggest changing the standardization routine names; even adding "standardize" at the beginning of them would be clearer.

All the `at` methods preform standardization and most of them viewpoint-adapt.  Only `atTypeDecl` does no viewpoint adaption.